sha2/src/aarch64.rs

```diff
@@ -0,0 +1,11 @@
+// From sys/auxv.h
+const AT_HWCAP: u64 = 16;
+const HWCAP_SHA2: u64 = 64;
```

Comment:
BTW is it possible that processor supports SHA-2 instructions, but not SHA-1 and/or AES? I thought they are part of the same `crypto` extension.

Reply:
The “crypto extension” is an ARM marketing term for a Cortex-A core bundling these three extensions, and possibly the sha512, sha3, and more in the future.

I’m not aware of any actual CPU exposing e.g. the sha2 flag but not sha1, but given they are different flags I’d rather test them properly.

Reply:
Since we are using `getauxval` it does not matter, but during migration to intrinsics we will have to use a more coarse grained target features, which only know about the `crypto` extension. But I guess since LLVM does not introduce separate finer-grained features it should be fine.

Reply:
How does it work for e.g. the sha3 flag?

Reply:
Honestly, I don't quite get how LLVM handles features, e.g. in [here](http://lists.llvm.org/pipermail/llvm-dev/2018-September/126346.html) they write:
>Extensions can now have different meanings based on the base architecture they apply to. For example on AArch64, 'crypto' means different things for v8.{1,2,3}-a than v8.4-a. The former adds 'sha2' and 'aes', the latter adds those and 'sm4' and 'sha3' on top.

I wonder why they use `crypto` instead of finer-grained `sha1`, `sha2`, `sha512`, `sha3` and `aes` features...

Reply:
They apparently noticed they made a mistake and now have aliased crypto to sha2 + aes, see https://github.com/llvm-mirror/llvm/commit/663484452ecf2000001548dfd28c5f816e2aa8b7#diff-5ff776706657bdcfc45aa7d7159bfeadR45

We should be able to use each feature separately nowadays.